cmd/juju/user/change_password.go

```diff
@@ -92,78 +120,126 @@ func (c *changePasswordCommand) Run(ctx *cmd.Context) error {
 		defer c.api.Close()
 	}
 
-	newPassword, err := readAndConfirmPassword(ctx)
-	if err != nil {
-		return errors.Trace(err)
+	if c.Reset {
+		return c.resetUserPassword(ctx)
 	}
+	return c.updateUserPassword(ctx)
+}
 
+func (c *changePasswordCommand) prepareRun() error {
 	controllerName, err := c.ControllerName()
 	if err != nil {
 		return errors.Trace(err)
 	}
-	store := c.ClientStore()
-	accountDetails, err := store.AccountDetails(controllerName)
+	c.controllerName = controllerName
```

Comment:
Sure as previously this was all part of Run, I guess it did not matter. I'll refactor into a separate method for maintainability.